### ISSUE TITLE
[BUG] [Segment Replication] checkpoints_behind in _cat/segment_replication API does not goes down to 0

### ISSUE THREAD

**Issue by dreamer-89** [opensearch-project/OpenSearch#8322]
**Describe the bug**
Coming from [bwc test](https://github.com/opensearch-project/OpenSearch/pull/7626) in 2.x branch, I see replica shard does not catch up with primary `checkpoints_behind` is always > 0. There are no active writes happening while waiting for relica to catch up. 

**To Reproduce**
Until there is better way to verify this. Run `testIndexingWithSegRep` bwc test on 2.x branch, the upgrade fails mid-way where primary stays on upgraded node while replica shard on non-upgraded node.

```
REPRODUCE WITH: ./gradlew ':qa:rolling-upgrade:v2.8.1#twoThirdsUpgradedTest' --tests "org.opensearch.upgrades.IndexingIT.testIndexingWithSegRep" -Dtests.seed=524BBE5736E9E0D4 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ro-RO -Dtests.timezone=Africa/Brazzaville -Druntime.java=17

org.opensearch.upgrades.IndexingIT > testIndexingWithSegRep FAILED
    java.lang.AssertionError: Replica shard [test-index-segrep][0]not upto date with primary  expected:<0> but was:<2>
        at __randomizedtesting.SeedInfo.seed([524BBE5736E9E0D4:DD32A81AC2B01515]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.opensearch.upgrades.IndexingIT.lambda$verifySegmentStats$1(IndexingIT.java:167)
        at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1082)
        at org.opensearch.upgrades.IndexingIT.verifySegmentStats(IndexingIT.java:158)
        at org.opensearch.upgrades.IndexingIT.waitForSearchableDocs(IndexingIT.java:102)
        at org.opensearch.upgrades.IndexingIT.testIndexingWithSegRep(IndexingIT.java:337)
```

#### Logs
Test logs depict that replica shard catches up on doc count but `checkpoints_behind` remains non-zero.
```

  1> [2023-06-28T05:00:46,224][INFO ][o.o.u.IndexingIT         ] [testIndexingWithSegRep] --> _cat/segments_replication
  1>  shardId                target_node checkpoints_behind
  1> [test-index-segrep][0] v2.8.1-2    2
  1> [test-index-segrep][0] v2.8.1-1    0
  1> [test-index-segrep][1] v2.8.1-1    0
  1> [test-index-segrep][1] v2.8.1-2    1
  1> [test-index-segrep][2] v2.8.1-0    0
  1> [test-index-segrep][2] v2.8.1-1    0

  1> [2023-06-28T05:00:46,229][INFO ][o.o.u.IndexingIT         ] [testIndexingWithSegRep] --> _cat/shards
  1> index             shard prirep state   docs  store ip        node
  1> test-index-segrep 2     r      STARTED    5  4.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 2     r      STARTED    5   230b 127.0.0.1 v2.8.1-1
  1> test-index-segrep 2     p      STARTED    5  4.3kb 127.0.0.1 v2.8.1-2
  1> test-index-segrep 1     p      STARTED    6  8.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 1     r      STARTED    6  8.3kb 127.0.0.1 v2.8.1-1
  1> test-index-segrep 1     r      STARTED    6  4.1kb 127.0.0.1 v2.8.1-2
  1> test-index-segrep 0     p      STARTED    5 12.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 0     r      STARTED    5 12.3kb 127.0.0.1 v2.8.1-1
  1> test-index-segrep 0     r      STARTED    4  4.2kb 127.0.0.1 v2.8.1-2

  1> [2023-06-28T05:00:46,236][INFO ][o.o.u.IndexingIT         ] [testIndexingWithSegRep] --> _cat/segments_replication
  1>  shardId                target_node checkpoints_behind
  1> [test-index-segrep][0] v2.8.1-2    2
  1> [test-index-segrep][0] v2.8.1-1    0
  1> [test-index-segrep][1] v2.8.1-1    0
  1> [test-index-segrep][1] v2.8.1-2    1
  1> [test-index-segrep][2] v2.8.1-1    0
  1> [test-index-segrep][2] v2.8.1-0    0

  1> [2023-06-28T05:00:46,239][INFO ][o.o.u.IndexingIT         ] [testIndexingWithSegRep] --> _cat/shards
  1> index             shard prirep state   docs  store ip        node
  1> test-index-segrep 2     r      STARTED    5  4.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 2     r      STARTED    5   230b 127.0.0.1 v2.8.1-1
  1> test-index-segrep 2     p      STARTED    5  4.3kb 127.0.0.1 v2.8.1-2
  1> test-index-segrep 1     p      STARTED    6  8.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 1     r      STARTED    6  8.3kb 127.0.0.1 v2.8.1-1
  1> test-index-segrep 1     r      STARTED    6  4.1kb 127.0.0.1 v2.8.1-2
  1> test-index-segrep 0     p      STARTED    5 12.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 0     r      STARTED    5 12.3kb 127.0.0.1 v2.8.1-1
  1> test-index-segrep 0     r      STARTED    4  4.2kb 127.0.0.1 v2.8.1-2

  1> [2023-06-28T05:00:46,247][INFO ][o.o.u.IndexingIT         ] [testIndexingWithSegRep] --> _cat/segments_replication
  1>  shardId                target_node checkpoints_behind
  1> [test-index-segrep][0] v2.8.1-2    2
  1> [test-index-segrep][0] v2.8.1-1    0
  1> [test-index-segrep][1] v2.8.1-1    0
  1> [test-index-segrep][1] v2.8.1-2    1
  1> [test-index-segrep][2] v2.8.1-1    0
  1> [test-index-segrep][2] v2.8.1-0    0

  1> [2023-06-28T05:00:46,251][INFO ][o.o.u.IndexingIT         ] [testIndexingWithSegRep] --> _cat/shards
  1> index             shard prirep state   docs  store ip        node
  1> test-index-segrep 2     r      STARTED    5  4.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 2     r      STARTED    5   230b 127.0.0.1 v2.8.1-1
  1> test-index-segrep 2     p      STARTED    5  4.3kb 127.0.0.1 v2.8.1-2
  1> test-index-segrep 1     p      STARTED    6  8.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 1     r      STARTED    6  8.3kb 127.0.0.1 v2.8.1-1
  1> test-index-segrep 1     r      STARTED    6  4.1kb 127.0.0.1 v2.8.1-2
  1> test-index-segrep 0     p      STARTED    5 12.3kb 127.0.0.1 v2.8.1-0
  1> test-index-segrep 0     r      STARTED    5 12.3kb 127.0.0.1 v2.8.1-1
  1> test-index-segrep 0     r      STARTED    5  4.2kb 127.0.0.1 v2.8.1-2
```

There are traces of `ActionNotFoundTransportException` exception on primary shard.
```
[2023-06-28T04:00:45,069][ERROR][o.o.i.r.SegmentReplicationTargetService] [v2.8.1-1] Failed to update visible checkpoint for replica [test-index-segrep][2], ReplicationCheckpoint{shardId=[test-index-segrep][2], primaryTerm=1, segmentsGen=4, version=9, size=230, codec=Lucene95}: RemoteTransportException[[v2.8.1-2][127.0.0.1:36499][internal:index/shard/replication/update_visible_checkpoint]]; nested: ActionNotFoundTransportException[No handler for action [internal:index/shard/replication/update_visible_checkpoint]];
...
[2023-06-28T04:00:46,183][ERROR][o.o.i.r.SegmentReplicationTargetService] [v2.8.1-1] Failed to update visible checkpoint for replica [test-index-segrep][2], ReplicationCheckpoint{shardId=[test-index-segrep][2], primaryTerm=1, segmentsGen=4, version=11, size=230, codec=Lucene95}: RemoteTransportException[[v2.8.1-2][127.0.0.1:36499][internal:index/shard/replication/update_visible_checkpoint]]; nested: ActionNotFoundTransportException[No handler for action [internal:index/shard/replication/update_visible_checkpoint]];
```

**Expected behavior**
The replica should eventually be caught up with primary and should be 0 checkpoints_behind.

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**
 - OS: ubuntu
 - Version 2.x

**Additional context**

**Comment by dreamer-89** [comment]
With https://github.com/opensearch-project/OpenSearch/pull/8020, the behavior around how primary tracks replica checkpoint was changed. The change was to rely on replica shard initiating a separate transport call to primary for checkpoint update when segment replication round finishes. This additional network call breaks backward compatibility with released versions where source does not have respective transport handler to update target replication checkpoint and fails with below error.

```
Failed to update visible checkpoint for replica [test-index-segrep][2], ReplicationCheckpoint{shardId=[test-index-segrep][2], primaryTerm=1, segmentsGen=4, version=9, size=230, codec=Lucene95}: RemoteTransportException[[v2.8.1-2][127.0.0.1:36499][internal:index/shard/replication/update_visible_checkpoint]]; nested: ActionNotFoundTransportException[No handler for action [internal:index/shard/replication/update_visible_checkpoint]];
```

**Comment by dreamer-89** [comment]
Opened a PR https://github.com/opensearch-project/OpenSearch/pull/8560 to add the logic back after discussion with @mch2. The PR though fails the bwc test because the primary shard running on bwc node (running on 2.9.0) expects the replica shard to make transport call for the update. In order to unblock this, I need to mute this test until this change is back-ported.

**Comment by dreamer-89** [comment]
Closing as #8560 is merged into main and backorted to 2.x
